examples/nucleo_g474re/sx128x_lora/main.cpp

```diff
@@ -0,0 +1,279 @@
+                    MODM_LOG_DEBUG << "RxTxTimeout Interrupt!" << modm::endl;
+                }
+                rx::dio3 = false;
+            }
+            PT_YIELD();
+        }
+        PT_END();
+    }
+
+private:
+    Irq_t irqStatus;
+    PacketStatus packetStatus;
+    RxBufferStatus rxBufferStatus;
+
+    modm::Sx128x< Hal > radio;
+
+private:
+
+    static constexpr uint8_t bufferSize = 124;
+    uint8_t buffer[bufferSize];
+} rxThread;
+
+class TxThread : public modm::sx128x, public modm::pt::Protothread
+{
+    using Reset = modm::platform::GpioC2;
+    using Busy = modm::platform::GpioC3;
+    using Dio1 = modm::platform::GpioA0;
+    using Dio2 = modm::platform::GpioA1;
+    using Dio3 = modm::platform::GpioA2;
+
+    using Nss = modm::platform::GpioC1;
+    using Hal = modm::Sx128xHalSpi<SpiMaster, Nss, Busy, Reset>;
+
+public:
+    TxThread() : timer(std::chrono::milliseconds(500)) {}
+
+    inline bool
+    run()
+    {
+        PT_BEGIN();
+
+        Nss::setOutput(modm::Gpio::High);
+        Reset::setOutput(modm::Gpio::Low);
+        Busy::setInput(modm::platform::Gpio::InputType::PullDown);
+
+        Dio1::setInput(modm::platform::Gpio::InputType::PullDown);
+        Exti::connect<Dio1>(Exti::Trigger::RisingEdge, [](uint8_t) mutable { tx::dio1 = true; });
+
+        Dio2::setInput(modm::platform::Gpio::InputType::PullDown);
+        Exti::connect<Dio2>(Exti::Trigger::RisingEdge, [](uint8_t) mutable { tx::dio2 = true; });
+
+        Dio3::setInput(modm::platform::Gpio::InputType::PullDown);
+        Exti::connect<Dio3>(Exti::Trigger::RisingEdge, [](uint8_t) mutable { tx::dio3 = true; });
+
+
+        radio.reset();
+        PT_CALL(radio.setStandby());
+
+        // Initialize the sx128x
+        PT_CALL(radio.setPacketType(PacketType::LoRa));
+        PT_CALL(radio.setRfFrequency(2457_MHz));
+        PT_CALL(radio.setRegulatorMode(RegulatorMode::Ldo));
+        PT_CALL(radio.setBufferBaseAddress(0, 0));
+        PT_CALL(radio.setModulationParams(modulationParams));
+        PT_CALL(radio.writeRegister(Register::SfAdditionalConfiguration, 0x32));
+        PT_CALL(radio.writeRegister(Register::FrequencyErrorCorrection, 0x01));
+        PT_CALL(radio.setPacketParams(packetParams));
+        PT_CALL(radio.setDioIrqParams(Irq::TxDone | Irq::RxTxTimeout, Irq::TxDone, Irq::RxTxTimeout));
+
+        MODM_LOG_DEBUG << "Sx128x initialization complete!" << modm::endl;
+
+        while (true)
+        {
+            if (tx::dio1)
+            {
+                PT_CALL(radio.getIrqStatus(irqStatus));
+                if (irqStatus.any(Irq::TxDone))
+                {
+                    PT_CALL(radio.clearIrqStatus(Irq::TxDone));
+                    irqStatus.reset(Irq::TxDone);
+
+                    MODM_LOG_DEBUG << "Message sent" << modm::endl;
+                    MODM_LOG_DEBUG << "Counter: " << counter << modm::endl;
+                    counter++;
+                }
+                tx::dio1 = false;
```

Comment:
That is a very good point! This could lead to some interesting behaviour in my own use case, so thanks for pointing it out. I have changed it to use `std::atomic_bool`.